File: VS2005/VB/BindingFiltering/DGAutoFilter/Form1.vb
'---------------------------------------------------------------------
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'---------------------------------------------------------------------

Option Strict On
Option Explicit On

Imports DataGridViewAutoFilter
Imports BindingFiltering

Public Class Form1

    Public Sub New()

        ' This call is required by the Windows Form Designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        Dim employees As FilteredBindingList(Of Employee) = New FilteredBindingList(Of Employee)
        employees.Add(New Employee("Aaberg", "Jesper", 52000, New DateTime(2004, 1, 14)))
        employees.Add(New Employee("Haas", "Jonathan", 75000, New DateTime(2003, 2, 14)))
        employees.Add(New Employee("Adams", "Ellen", 52000, New DateTime(2006, 10, 16)))
        employees.Add(New Employee("Hanif", "Kerim", 60000, New DateTime(2005, 10, 4)))
        employees.Add(New Employee("Akers", "Kim", 55000, New DateTime(2004, 1, 14)))
        employees.Add(New Employee("Harris", "Phyllis", 66000, New DateTime(2007, 3, 10)))
        employees.Add(New Employee("Andersen", "Elizabeth", 55000, New DateTime(2003, 7, 10)))
        employees.Add(New Employee("Alberts", "Amy", 40000, New DateTime(2005, 7, 7)))
        employees.Add(New Employee("Hamlin", "Jay", 45000, New DateTime(2005, 8, 8)))
        employees.Add(New Employee("Hee", "Gordon", 35000, New DateTime(2004, 9, 9)))
        employees.Add(New Employee("Penor", "Lori", 52000, New DateTime(2004, 11, 11)))
        employees.Add(New Employee("Pfeiffer", "Michael", 51000, New DateTime(2006, 12, 12)))
        BindingSource1.DataSource = employees
        DataGridView1.DataSource = BindingSource1

        showAllLabel.Text = "Show &All"
        showAllLabel.Visible = False
        showAllLabel.IsLink = True
        showAllLabel.LinkBehavior = LinkBehavior.HoverUnderline
    End Sub

    Private Sub DataGridView1_DataBindingComplete(ByVal sender As Object, ByVal e As System.Windows.Forms.DataGridViewBindingCompleteEventArgs) Handles DataGridView1.DataBindingComplete
        Dim filterStatus As String = DataGridViewAutoFilterColumnHeaderCell _
            .GetFilterStatus(DataGridView1)
        If (String.IsNullOrEmpty(filterStatus)) Then
            showAllLabel.Visible = False
            filterStatusLabel.Visible = False
        Else
            showAllLabel.Visible = True
            filterStatusLabel.Visible = True
            filterStatusLabel.Text = filterStatus
        End If
    End Sub

    Private Sub showAllLabel_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles showAllLabel.Click
        DataGridViewAutoFilterTextBoxColumn.RemoveFilter(DataGridView1)
    End Sub

    Private Sub DataGridView1_BindingContextChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles DataGridView1.BindingContextChanged
        ' Continue only if the data source has been set.
        If (DataGridView1.DataSource Is Nothing) Then
            Return
        End If

        ' Add the AutoFilter header cell to each column.
        Dim col As DataGridViewColumn
        For Each col In DataGridView1.Columns
            col.HeaderCell = New _
                DataGridViewAutoFilterColumnHeaderCell(col.HeaderCell)
        Next

        ' Resize the columns to fit their contents.
        DataGridView1.AutoResizeColumns()
    End Sub

End Class

File: VS2005/CS/BindingFiltering/DGAutoFilter/Form1.cs
//---------------------------------------------------------------------
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//---------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DataGridViewAutoFilter;
using BindingFiltering;


namespace DGAutoFilter
{
    public partial class Form1 : Form
    {
        private DataGridView dataGridView1;
        private BindingSource bindingSource1;
        private StatusStrip statusStrip1;
        private ToolStripStatusLabel showAllLabel;
        private ToolStripStatusLabel filterStatusLabel;
        private IContainer components;

        public Form1()
        {
            InitializeComponent();
            FilteredBindingList<Employee> employees = new FilteredBindingList<Employee>();
            employees.Add(new Employee("Aaberg", "Jesper", 52000, new DateTime(2004, 1, 14)));
            employees.Add(new Employee("Haas", "Jonathan", 75000, new DateTime(2003, 2, 14)));
            employees.Add(new Employee("Adams", "Ellen", 52000, new DateTime(2006, 10, 16)));
            employees.Add(new Employee("Hanif", "Kerim", 60000, new DateTime(2005, 10, 4)));
            employees.Add(new Employee("Akers", "Kim", 55000, new DateTime(2004, 1, 14)));
            employees.Add(new Employee("Harris", "Phyllis", 66000, new DateTime(2007, 3, 10)));
            employees.Add(new Employee("Andersen", "Elizabeth", 55000, new DateTime(2003, 7, 10)));
            employees.Add(new Employee("Alberts", "Amy", 40000, new DateTime(2005, 7, 7)));
            employees.Add(new Employee("Hamlin", "Jay", 45000, new DateTime(2005, 8, 8)));
            employees.Add(new Employee("Hee", "Gordon", 35000, new DateTime(2004, 9, 9)));
            employees.Add(new Employee("Penor", "Lori", 52000, new DateTime(2004, 11, 11)));
            employees.Add(new Employee("Pfeiffer", "Michael", 51000, new DateTime(2006, 12, 12)));
            bindingSource1.DataSource = employees;
            dataGridView1.DataSource = bindingSource1;
            dataGridView1.BindingContextChanged += new EventHandler(dataGridView1_BindingContextChanged);

            showAllLabel.Text = "Show &All";
            showAllLabel.Visible = false;
            showAllLabel.Click += new EventHandler(showAllLabel_Click);
            showAllLabel.IsLink = true;
            showAllLabel.LinkBehavior = LinkBehavior.HoverUnderline;

            dataGridView1.DataBindingComplete += new DataGridViewBindingCompleteEventHandler(dataGridView1_DataBindingComplete);

        }

        void dataGridView1_DataBindingComplete(object sender,
    DataGridViewBindingCompleteEventArgs e)
        {
            String filterStatus = DataGridViewAutoFilterColumnHeaderCell
                .GetFilterStatus(dataGridView1);
            if (String.IsNullOrEmpty(filterStatus))
            {
                showAllLabel.Visible = false;
                filterStatusLabel.Visible = false;
            }
            else
            {
                showAllLabel.Visible = true;
                filterStatusLabel.Visible = true;
                filterStatusLabel.Text = filterStatus;
            }
        }


        private void showAllLabel_Click(object sender, EventArgs e)
        {
            DataGridViewAutoFilterTextBoxColumn.RemoveFilter(dataGridView1);
        }


        void dataGridView1_BindingContextChanged(object sender, EventArgs e)
        {
            // Continue only if the data source has been set.
            if (dataGridView1.DataSource == null)
            {
                return;
            }

            // Add the AutoFilter header cell to each column.
            foreach (DataGridViewColumn col in dataGridView1.Columns)
            {
                col.HeaderCell = new
                    DataGridViewAutoFilterColumnHeaderCell(col.HeaderCell);
            }

            // Resize the columns to fit their contents.
            dataGridView1.AutoResizeColumns();


        }

        private void InitializeComponent()
        {
            this.components = new System.ComponentModel.Container();
            System.Windows.Forms.DataGridViewCellStyle dataGridViewCellStyle1 = new System.Windows.Forms.DataGridViewCellStyle();
            System.Windows.Forms.DataGridViewCellStyle dataGridViewCellStyle2 = new System.Windows.Forms.DataGridViewCellStyle();
            System.Windows.Forms.DataGridViewCellStyle dataGridViewCellStyle3 = new System.Windows.Forms.DataGridViewCellStyle();
            this.dataGridView1 = new System.Windows.Forms.DataGridView();
            this.bindingSource1 = new System.Windows.Forms.BindingSource(this.components);
            this.statusStrip1 = new System.Windows.Forms.StatusStrip();
            this.showAllLabel = new System.Windows.Forms.ToolStripStatusLabel();
            this.filterStatusLabel = new System.Windows.Forms.ToolStripStatusLabel();
            ((System.ComponentModel.ISupportInitialize)(this.dataGridView1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.bindingSource1)).BeginInit();
            this.statusStrip1.SuspendLayout();
            this.SuspendLayout();
            // 
            // dataGridView1
            // 
            dataGridViewCellStyle1.Alignment = System.Windows.Forms.DataGridViewContentAlignment.MiddleLeft;
            dataGridViewCellStyle1.BackColor = System.Drawing.SystemColors.Control;
            dataGridViewCellStyle1.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            dataGridViewCellStyle1.ForeColor = System.Drawing.SystemColors.WindowText;
            dataGridViewCellStyle1.SelectionBackColor = System.Drawing.SystemColors.Highlight;
            dataGridViewCellStyle1.SelectionForeColor = System.Drawing.SystemColors.HighlightText;
            dataGridViewCellStyle1.WrapMode = System.Windows.Forms.DataGridViewTriState.True;
            this.dataGridView1.ColumnHeadersDefaultCellStyle = dataGridViewCellStyle1;
            this.dataGridView1.ColumnHeadersHeightSizeMode = System.Windows.Forms.DataGridViewColumnHeadersHeightSizeMode.AutoSize;
            dataGridViewCellStyle2.Alignment = System.Windows.Forms.DataGridViewContentAlignment.MiddleLeft;
            dataGridViewCellStyle2.BackColor = System.Drawing.SystemColors.Window;
            dataGridViewCellStyle2.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            dataGridViewCellStyle2.ForeColor = System.Drawing.SystemColors.ControlText;
            dataGridViewCellStyle2.SelectionBackColor = System.Drawing.SystemColors.Highlight;
            dataGridViewCellStyle2.SelectionForeColor = System.Drawing.SystemColors.HighlightText;
            dataGridViewCellStyle2.WrapMode = System.Windows.Forms.DataGridViewTriState.False;
            this.dataGridView1.DefaultCellStyle = dataGridViewCellStyle2;
            this.dataGridView1.Location = new System.Drawing.Point(23, 45);
            this.dataGridView1.Name = "dataGridView1";
            dataGridViewCellStyle3.Alignment = System.Windows.Forms.DataGridViewContentAlignment.MiddleLeft;
            dataGridViewCellStyle3.BackColor = System.Drawing.SystemColors.Control;
            dataGridViewCellStyle3.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            dataGridViewCellStyle3.ForeColor = System.Drawing.SystemColors.WindowText;
            dataGridViewCellStyle3.SelectionBackColor = System.Drawing.SystemColors.Highlight;
            dataGridViewCellStyle3.SelectionForeColor = System.Drawing.SystemColors.HighlightText;
            dataGridViewCellStyle3.WrapMode = System.Windows.Forms.DataGridViewTriState.True;
            this.dataGridView1.RowHeadersDefaultCellStyle = dataGridViewCellStyle3;
            this.dataGridView1.Size = new System.Drawing.Size(442, 340);
            this.dataGridView1.TabIndex = 0;
            // 
            // statusStrip1
            // 
            this.statusStrip1.Dock = System.Windows.Forms.DockStyle.Top;
            this.statusStrip1.Items.AddRange(new System.Windows.Forms.ToolStripItem[] {
            this.showAllLabel,
            this.filterStatusLabel});
            this.statusStrip1.Location = new System.Drawing.Point(0, 0);
            this.statusStrip1.Name = "statusStrip1";
            this.statusStrip1.Size = new System.Drawing.Size(548, 22);
            this.statusStrip1.TabIndex = 1;
            this.statusStrip1.Text = "statusStrip1";
            // 
            // showAllLabel
            // 
            this.showAllLabel.Name = "showAllLabel";
            this.showAllLabel.Size = new System.Drawing.Size(53, 17);
            this.showAllLabel.Text = "Show All";
            // 
            // filterStatusLabel
            // 
            this.filterStatusLabel.Name = "filterStatusLabel";
            this.filterStatusLabel.Size = new System.Drawing.Size(39, 17);
            this.filterStatusLabel.Text = "Status";
            // 
            // Form1
            // 
            this.ClientSize = new System.Drawing.Size(548, 414);
            this.Controls.Add(this.statusStrip1);
            this.Controls.Add(this.dataGridView1);
            this.Name = "Form1";
            ((System.ComponentModel.ISupportInitialize)(this.dataGridView1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.bindingSource1)).EndInit();
            this.statusStrip1.ResumeLayout(false);
            this.statusStrip1.PerformLayout();
            this.ResumeLayout(false);
            this.PerformLayout();
        }
    }
}
